pkg/printers/internalversion/printers.go

```diff
@@ -1409,14 +1410,29 @@ func printStatefulSet(obj *apps.StatefulSet, options printers.GenerateOptions) (
 	desiredReplicas := obj.Spec.Replicas
 	readyReplicas := obj.Status.ReadyReplicas
 	createTime := translateTimestampSince(obj.CreationTimestamp)
-	row.Cells = append(row.Cells, obj.Name, fmt.Sprintf("%d/%d", int64(readyReplicas), int64(desiredReplicas)), createTime)
+	status := statefulSetStatus(obj)
+	row.Cells = append(row.Cells, obj.Name, fmt.Sprintf("%d/%d", int64(readyReplicas), int64(desiredReplicas)), status, createTime)
 	if options.Wide {
 		names, images := layoutContainerCells(obj.Spec.Template.Spec.Containers)
 		row.Cells = append(row.Cells, names, images)
 	}
 	return []metav1.TableRow{row}, nil
 }
 
+func statefulSetStatus(obj *apps.StatefulSet) string {
+	switch {
+	case obj.ObjectMeta.DeletionTimestamp != nil:
+		return "Terminating"
+	case obj.Status.CurrentRevision != obj.Status.UpdateRevision:
+		return "Updating"
+	case obj.Status.Replicas != obj.Spec.Replicas:
+		return "Scaling"
+	case obj.Status.ObservedGeneration != nil && *obj.Status.ObservedGeneration == obj.Generation:
+		return "Ready"
+	}
+	return "Unknown"
+}
```

Comment:
We don't expose that kind of information on a StatefulSet nor on any other workload controller. I'd suggest picking a different, more meaningful information. The proposed above states are not quite descriptive, but actually can be misleading.

Reply:
These were my thoughts too. Do you have suggestions for meaningful status information?

Reply:
hi @brianpursley @mpuckett159, any suggestions about it?